lib/cext/patches/racc_patches.rb

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative 'common_patches'
+
+class RaccPatches < CommonPatches
+
+  CPARSE_PARAM_VALUE_FIELDS = %w[value_v parser lexer action_table action_check action_default action_pointer
+                                 goto_table goto_check goto_default goto_pointer reduce_table token_table state
+                                 vstack tstack retval]
+
+  def self.generate_cparse_param_handles
+    CPARSE_PARAM_VALUE_FIELDS.collect do |field|
+      read_write_field('v', field, true)
+    end.flatten
+  end
+
+  PATCHES = {
+    gem: 'racc',
+    patches: {
+      'cparse.c' => [
+        { # Convert fixnum constants to ints because we convert cparse_params.t to an int.
+            match: /INT2FIX\((\w+?_TOKEN)\)/,
+            replacement: '\\1'
+        },
+        { # Change signature because when we call the callback we're not providing a self value -- difference from MRI.
+            match: 'static VALUE reduce0 _((VALUE block_args, VALUE data, VALUE self));',
+            replacement: 'static VALUE reduce0 _((VALUE block_args, VALUE data));'
+        },
+        { # Change signature because when we call the callback we're not providing a self value -- difference from MRI.
```

Comment:
Sounds like something that would be better fixed in ruby.c, or is there a reason we don't do it like MRI?

Reply:
I don't know. I need to debug MRI to see what's going on. This parameter is never used. I can't tell if a `Qnil` is automatically being inserted or what. We seem to be doing the correct thing.